Symbol display incorrect for CS0118 with parameters

```
using System;
public class Program
{
    static void Main(params string[] args)
    {
        Action<args> a = null;
    }
}
```
Error:
```
error CS0118: 'params string[]' is a variable but is used like a type
```
Should be `args` and not `params string[]` (works without `params` too, just added to show the effect)

Note: works fine for locals, just seems to be issue with parameters.

--- a/src/Compilers/CSharp/Portable/Binder/LookupResult.cs
+++ b/src/Compilers/CSharp/Portable/Binder/LookupResult.cs
@@ -191,7 +191,7 @@ namespace Microsoft.CodeAnalysis.CSharp
         internal static SingleLookupResult NotTypeOrNamespace(Symbol unwrappedSymbol, Symbol symbol, bool diagnose)
         {
             // TODO: determine correct diagnosis 
-            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
+            var diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadSKknown, unwrappedSymbol.Name, unwrappedSymbol.GetKindText(), MessageID.IDS_SK_TYPE.Localize()) : null;
             return new SingleLookupResult(LookupResultKind.NotATypeOrNamespace, symbol, diagInfo);
         }
 

--- a/src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/Semantics/SemanticErrorTests.cs
@@ -2239,6 +2239,28 @@ public class MyClass {
         }
 
         [Fact]
+        public void CS0118ERR_BadSKknown_02()
+        {
+            CreateCompilationWithMscorlib(@"
+using System;
+public class P {
+    public static void Main(string[] args) {
+#pragma warning disable 219
+        Action<args> a = null;
+        Action<a> b = null;
+    }
+}")
+                .VerifyDiagnostics(
+                    // (6,16): error CS0118: 'args' is a variable but is used like a type
+                    //         Action<args> a = null;
+                    Diagnostic(ErrorCode.ERR_BadSKknown, "args").WithArguments("args", "variable", "type").WithLocation(6, 16),
+                    // (7,16): error CS0118: 'a' is a variable but is used like a type
+                    //         Action<a> b = null;
+                    Diagnostic(ErrorCode.ERR_BadSKknown, "a").WithArguments("a", "variable", "type").WithLocation(7, 16));
+        }
+
+
+        [Fact]
         public void CS0118ERR_BadSKknown_CheckedUnchecked()
         {
             string source = @"

